packages/ui-lib/src/Misc/molecules/Pagination.tsx

```diff
@@ -0,0 +1,351 @@
+    onPageChange,
+    onItemsChange,
+  } = props;
+
+  const [fieldState, setFieldState] = useState<string>('default');
+  const [pageValue, setPageValue] = useState<string>(activePage.toString());
+  const [disableGO, setDisabledGo] = useState<boolean>(parseInt(pageValue) > totalPages && fieldState !== '' ? false : true);
+  const [shouldShake, setShouldShake] = useState<boolean>(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const getValidWidth = useCallback(() => {
+    return `${(totalPages.toString().length * WIDTH_PER_NUMBER) + WIDTH_PER_NUMBER / 2}px`;
+  }, [totalPages]);
+
+
+  const isValidInput = useCallback((value: string) => {
+
+    if (isNotNumber(value)) {
+      return false;
+    }
+
+    if (parseInt(value) > totalPages) {
+      return false;
+    }
+
+    if (parseInt(value) <= 0) {
+      return false;
+    }
+
+    if (value === '') {
+      return false;
+    }
+
+    return true;
+  }, [totalPages]);
+
+
+  const onInputChange = useCallback(({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+
+    setShouldShake(false);
+    if (isNotNumber(value)) {
+      return;
+    }
+
+    // max total digits, without left zeros or empty value
+    const validDigitValue = value === '' ? '' : Number(value.slice(-totalPages.toString().length)).toString();
+
+    setPageValue(validDigitValue);
+    if (isValidInput(validDigitValue)) {
+      setFieldState('processing');
+      setDisabledGo(false);
+
+    } else {
+      setFieldState('invalid');
+      setDisabledGo(true);
+      setShouldShake(true);
+    }
+
+  }, [isValidInput, totalPages]);
+
+
+  const onBlur = useCallback(({ target: { value } }: React.FocusEvent<HTMLInputElement>) => {
+
+    if ((value === '')) {
+      setPageValue(activePage.toString());
+
+    } else if (isValidInput(value) && parseInt(value) !== activePage) {
+      setDisabledGo(false);
+      setFieldState('processing');
+      return;
+
+    } else if (!isValidInput(value)) {
+      setFieldState('invalid');
+      setDisabledGo(true);
+      return;
+    }
+
+    setFieldState('default');
+
+  }, [activePage, isValidInput]);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+
+    if ((event.key === 'Enter') && isValidInput(pageValue)) {
+      onClickGo();
+    }
+
+  };
+
+  /**
+   * Review if current edith is valid if not make disable
+   */
+  const onFocus = ({ target: { value } }: React.FocusEvent<HTMLInputElement>) => {
+
+    if (isValidInput(value)) {
+      setFieldState('processing');
+    } else {
+      setFieldState('invalid');
+      setDisabledGo(true);
+    }
+
+  };
+
+  const getTextColor = useCallback((state: string): string => {
+
+    switch (state) {
+      case 'processing':
+        return 'var(--input-border-processing, #4db5ff)';
+        break;
+
+      case 'invalid':
+        return 'var(--input-border-invalid, #f66)';
+
+      case 'default':
+      default:
+        return 'var(--input-border-default, --grey-9)';
```

Comment:
`grey-9` is suitable here but the `--input-border-default` is the wrong color for this and should be removed.